CDCgov/prime-reportstream
# 8696 RADx MARS conditional receiver transforms

This PR includes the conditionals needed to support the RADx MARS HL7 v2 format. The RADx format expects Observations to appear in the following order: The COVID-19 result should be the first OBX segment, followed by any other result OBX segments, and then the Age, Sympotmatic, and Date of Onset OBX segments. A follow up ticket to create an Observation based on the patient's age has been created [here](https://app.zenhub.com/workspaces/platform-6182b02547c1130010f459db/issues/gh/cdcgov/prime-reportstream/9533).

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:

1. Send a test report using the attached Postman collection
[RADx MARS.postman_collection.txt](https://github.com/CDCgov/prime-reportstream/files/11520229/RADx.MARS.postman_collection.txt)


2. Verify that the request was accepted
3. Open the generated HL7 file in the local sftp folder.
4. Verify the RADx MARS conditionals listed in column H  in the spreadsheet [here](https://docs.google.com/spreadsheets/d/15dKu5zSY23-VVL0oJphILQyHjyDoj_WmTmXUxGXinzQ/edit#gid=0) appear on the HL7 message
5. Verify that OBX segments appear in this order: The COVID-19 result should be the first OBX segment, followed by any other result OBX segments, Sympotmatic, and Date of Onset OBX segments.
6. Run integration tests and make sure they pass 

## Changes
- Added RADx MARS receiver transforms
- Added integration tests for the RADx MARS format

## Checklist

### Testing

### Process

## Linked Issues
- #8696 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #9531 
- #9532 
- #9533 
- #9623 

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
